### DISCUSSION TITLE
Additional test database slots premium

### DISCUSSION THREAD

**Discussion by Godje** [monkeytypegame/monkeytype#1074] [Ideas] [answered]
TLDR: Pay for additional test database slots.

So far certain account's amount of detailed tests saved is 1000. Considering that I go about 30 tests a day, that's about a month worth of results. Monkeytype's graph system and everything about it is very informative and appealing. Since people might have multiple tags, some tags might simply lose results after one month.

I assume the limitation is also technical. Sending more than 1000 data points and rendering them results in bad experience.

I propose a paid feature, where a person can pay $10 or something, to increase the cap to 10 thousand. The burden of rendering or caching that many datapoints should be layed off to the client.

Or simply add an option to upload/download a CSV of results to see a nice graph.

This is just an example, and the pricing system can be whatever you want.

**Comment by Miodec** [comment, marked as answer]
Yeap, in the works. Patreon integration is pretty much almost done.

**Reply by Godje** [reply]
Hmmm, I wish it was more of a one time purchase type thing.
Like, buying extra slots for money like $1 per 1k or more

this way patreon is not ticking money away every month

Also, time comes where you are no longer going to be a patreon, what happens to the slots then? All the progress that you were paying for just disappears?

**Reply by Miodec** [reply]
The results are always stored in the database, so if you stop being a patron you go back to the 1000 limit. All the results are still stored though.

As for the monthly thing - my patreon is set to charge up front, so you could pay the dollar and cancel. Integrating with the patreon API is currently the easiest way of actually adding this system and since I've got no experience with implementing payment into a website I don't want to mess anything up.

**Reply by Godje** [reply]
Lets say I have 10000 limit and a patreon. 
I have racked up 10000 test results.
I stop paying for Patreon. Limit goes back to 1k
if I take one more test, the other 9k of tests are hidden now, until I start paying again?

**Reply by Miodec** [reply]
Yes

**Reply by Godje** [reply]
Doesn't sound sound to me lol. If I have those results why cut me from seeing them?

And if I stop paying, just keep cutting from the end like it usually does it. My impact on the database doesn't change whether I pay or not, then what's the point?

**Reply by Godje** [reply]
Not to be an arrogant prick who wants things for free or something. Just saying that it doesn't really make sense to me to have that limitation, when the impact of it isn't there. Unless I am missing something crucial in the architecture.

**Reply by Miodec** [reply]
Firebase firestore charges per document read. Each result is a document. If I allowed users to read all of their results the cost would snowball very fast.

**Reply by Godje** [reply]
I assume your server side deployment is closed source?

**Reply by Godje** [reply]
?

**Reply by Miodec** [reply]
Its not

**Reply by CodeMonkey80s** [reply]
Any ETA for this feature?

**Reply by Miodec** [reply]
Im ill right now so just taking some time off meaning most things are delayed. No eta's - im bad at them :P

**Reply by CodeMonkey80s** [reply]
get well!

**Reply by unrealapex** [reply]
Hope you get well soon!

**Reply by fehmer** [reply]
hi @Miodec , is this available now?

**Reply by Miodec** [reply]
No, sorry. I got caught up in a lot of changes since then, and the patreon integration got put aside. No eta's sorry.

**Reply by bchooxg** [reply]
Hi i was recently searching up for any solutions to this issue. I do really want to see the number of tests saved to be increased. Hoping that it gets picked up :)
